diff --git a/src/nixos-anywhere.sh b/src/nixos-anywhere.sh
index e5d713e..2cbca5c 100755
--- a/src/nixos-anywhere.sh
+++ b/src/nixos-anywhere.sh
@@ -11,6 +11,7 @@ enableDebug=""
 nixBuildFlags=()
 diskoScript=""
 diskoMode="disko"
+partitionScript=""
 nixosSystem=""
 extraFiles=""
 vmTest="n"
@@ -253,6 +254,10 @@ parseArgs() {
         ;;
       esac
 
+      shift
+      ;;
+    --partition-script)
+      partitionScript=$2
       shift
       ;;
     --build-on)
@@ -843,7 +848,12 @@ main() {
   fi
 
   if [[ ${phases[disko]} == 1 ]]; then
-    runDisko "$diskoScript"
+    if [[ -z $partitionScript ]]; then
+      runDisko "$diskoScript"
+    else
+      step Formatting hard drive with the partition script
+      runSsh "$partitionScript"
+    fi
   fi
 
   if [[ ${phases[install]} == 1 ]]; then
